EIPS/eip-4762.md

```diff
@@ -39,47 +39,44 @@ We define access events as follows. When an access event takes place, the access
 
 #### Access events for account headers
 
-When a non-precompile address is the target of a `CALL`, `CALLCODE`, `DELEGATECALL`, `SELFDESTRUCT`, `EXTCODESIZE`, or `EXTCODECOPY` opcode, or is the target address of a contract creation whose initcode starts execution, process these access events:
+When:
 
-```
-(address, 0, VERSION_LEAF_KEY)
-(address, 0, CODE_SIZE_LEAF_KEY)
-```
+ 1. a non-precompile address is the target of a `*CALL`, `SELFDESTRUCT`, `EXTCODESIZE`, or `EXTCODECOPY` opcode, 
+ 2. a non-precompile address is the target address of a contract creation whose initcode starts execution,
+ 3. any address is the target of the `BALANCE` opcode
+ 4. a _deployed_ contract calls `CODECOPY`
 
-If a call is value-bearing (ie. it transfers nonzero wei), whether or not the callee is a precompile, process these two access events:
+process this access events:
 
 ```
-(caller_address, 0, BALANCE_LEAF_KEY)
-(callee_address, 0, BALANCE_LEAF_KEY)
+(address, 0, BASIC_DATA_LEAF_KEY)
 ```
 
-When a contract is created, process these access events:
+If a `*CALL` or `SELFDESTRUCT` is value-bearing (ie. it transfers nonzero wei), whether or not the callee is a precompile, process these additional two access events:
 
 ```
-(contract_address, 0, VERSION_LEAF_KEY)
-(contract_address, 0, NONCE_LEAF_KEY)
-(contract_address, 0, BALANCE_LEAF_KEY)
-(contract_address, 0, CODE_KECCAK_LEAF_KEY)
-(contract_address, 0, CODE_SIZE_LEAF_KEY)
+(caller_address, 0, BASIC_DATA_LEAF_KEY)
+(callee_address, 0, BASIC_DATA_LEAF_KEY)
 ```
 
-If the `BALANCE` opcode is called targeting some address, process this access event:
+When calling `EXTCODEHASH`, process the access event:
```

Comment:
@gballet, I think we should add here "if the target isn't a precompile".

Reply:
I don't think so: looking at 2929, there is no check whether the target is a precompile or not.

Reply:
I believe EIP-2929 rules are because you're supposed to load the whole account in one go.

What this rule is saying in this EIP is that adding a precompile codehash to the witness makes sense, which I think it doesn't? In other parts of this EIP we have rules that exclude precompiles.

Reply:
```suggestion
When calling `EXTCODEHASH` on a contract that is not a precompile, process the access event:
```

Reply:
TODO: add rationale for this